### ISSUE TITLE
Watch failures on Quarkus native: Caused by: com.fasterxml.jackson.databind.JsonMappingException: No resource type found for:v1#Status

### ISSUE THREAD

**Issue by goldmann** [java-operator-sdk/java-operator-sdk#284]
When running Quarkus app using the SDK in **native** mode, such stacktraces as pasted below are present in the log.

These are printed **every second** for every single CustomResource watched by the SDK (my assumption). In my case I have two CR's and two such stacktraces in the log every second (see below).

Such stacktraces do not appear immediately after running the app, it takes a while for them to appear. Additionally, it looks like memory usage goes up and up.

Possibly related: https://github.com/fabric8io/kubernetes-client/issues/2244

```
2021-01-05 08:32:40,823 WARN  [io.fab.kub.cli.dsl.int.WatchConnectionManager] (OkHttp https://172.30.0.1/...) Exec Failure: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:249)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:168)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:153)
	at io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager.readWatchEvent(WatchHTTPManager.java:292)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onMessage(WatchConnectionManager.java:228)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:322)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:219)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:273)
	at okhttp3.internal.ws.RealWebSocket$1.onResponse(RealWebSocket.java:209)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:174)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No resource type found for:v1#Status
 at [Source: (BufferedInputStream); line: 1, column: 180] (through reference chain: io.fabric8.kubernetes.api.model.WatchEvent["object"])
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.fromObjectNode(KubernetesDeserializer.java:85)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:57)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:46)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:156)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4526)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3513)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:247)
	... 16 more

2021-01-05 08:32:40,942 WARN  [io.fab.kub.cli.dsl.int.WatchConnectionManager] (OkHttp https://172.30.0.1/...) Exec Failure: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:249)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:168)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:153)
	at io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager.readWatchEvent(WatchHTTPManager.java:292)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onMessage(WatchConnectionManager.java:228)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:322)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:219)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:273)
	at okhttp3.internal.ws.RealWebSocket$1.onResponse(RealWebSocket.java:209)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:174)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No resource type found for:v1#Status
 at [Source: (BufferedInputStream); line: 1, column: 180] (through reference chain: io.fabric8.kubernetes.api.model.WatchEvent["object"])
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.fromObjectNode(KubernetesDeserializer.java:85)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:57)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:46)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:156)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4526)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3513)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:247)
	... 16 more

2021-01-05 08:32:41,873 WARN  [io.fab.kub.cli.dsl.int.WatchConnectionManager] (OkHttp https://172.30.0.1/...) Exec Failure: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:249)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:168)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:153)
	at io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager.readWatchEvent(WatchHTTPManager.java:292)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onMessage(WatchConnectionManager.java:228)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:322)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:219)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:273)
	at okhttp3.internal.ws.RealWebSocket$1.onResponse(RealWebSocket.java:209)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:174)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No resource type found for:v1#Status
 at [Source: (BufferedInputStream); line: 1, column: 180] (through reference chain: io.fabric8.kubernetes.api.model.WatchEvent["object"])
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.fromObjectNode(KubernetesDeserializer.java:85)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:57)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:46)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:156)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4526)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3513)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:247)
	... 16 more

2021-01-05 08:32:41,992 WARN  [io.fab.kub.cli.dsl.int.WatchConnectionManager] (OkHttp https://172.30.0.1/...) Exec Failure: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:249)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:168)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:153)
	at io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager.readWatchEvent(WatchHTTPManager.java:292)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onMessage(WatchConnectionManager.java:228)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:322)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:219)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:273)
	at okhttp3.internal.ws.RealWebSocket$1.onResponse(RealWebSocket.java:209)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:174)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No resource type found for:v1#Status
 at [Source: (BufferedInputStream); line: 1, column: 180] (through reference chain: io.fabric8.kubernetes.api.model.WatchEvent["object"])
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.fromObjectNode(KubernetesDeserializer.java:85)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:57)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:46)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:156)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4526)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3513)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:247)
	... 16 more
```

**Comment by goldmann** [comment]
![image](https://user-images.githubusercontent.com/43489/103625729-abdd3d80-4f3b-11eb-8e36-e001c3daeca2.png)

Memory usage according to time.

**Comment by goldmann** [comment]
Ok, this may be a PEBKAC in fact, here are the things that I applied and I observe no more stacktraces:

1. The `*Spec` and `*Status` classes for my `CustomResource`s did not implement `KubernetesResource` - I've fixed it.
2. I was missing a setter on the `*Spec` and `*Status` objects in my `CustomResource`s - I've fixed it.
3. I've added `@JsonDeserialize(using = JsonDeserializer.None.class)` annotation to the `*Spec` and `*Status` classes as suggested by the examples found here: https://github.com/fabric8io/kubernetes-client/tree/v4.13.0/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/crd

With these changes the app is now running for over 8h without throwing any exceptions. Earlier it started throwing them about 1hr from start. I'm not exactly sure which of the above changed the behavior.

Saying this, without them, the app worked as well, just not very stable which is a bit confusing tbh.

Mentioned memory issue is probably not related to this at all, I suspect that it may be related to caching events received as part of my event sources implementation.

I'm closing the ticket for now.
